redis/tests/test_async.rs

```diff
@@ -644,15 +644,20 @@ mod basic_async {
         let mut con = ctx.multiplexed_async_connection().await.unwrap();
         for i in 0..20usize {
             let _: () = con.append(format!("test/{i}"), i).await.unwrap();
+            let _: () = con.append(format!("other/{i}"), i).await.unwrap();
         }
-        let opts = ScanOptions::default().count(20).pattern("test/*");
+        let opts = ScanOptions::default().set_count(20).set_pattern("test/*");
         let values = con.scan_options::<String>(opts).await.unwrap();
         let values: Vec<_> = timeout(Duration::from_millis(100), values.collect())
```

Comment:
just to make the tests clear - add here the doc "scan with pattern", and below "scan without pattern", just so it'll be easy to parse what's the difference between the 2 queries.